### PR TITLE
Meson packaging

### PR DESCRIPTION
**Pull request by SoapGentoo**
@mr-c some canonical Meson cleanups, in preparation for your PR in https://github.com/PacificBiosciences/pbcopper/pull/11

### CODE REVIEW

**Comment by mr-c** [comment]
@SoapGentoo Super cool! I'll let @nemequ review this one.

File: meson.build [hunk 1]
```diff
@@ -6,31 +6,132 @@ project('SIMDe', 'c', 'cpp',
...
-		'simde/x86/mmx.h',
-		'simde/x86/sse.h',
-		'simde/x86/sse2.h',
-		'simde/x86/sse3.h',
-		'simde/x86/sse4.1.h',
-		'simde/x86/sse4.2.h',
-		'simde/x86/ssse3.h',
-		'simde/x86/svml.h',
-    ],
+if not meson.is_subproject() and get_option('tests')
```
Comment (nemequ) [inline meson.build:9]: If it's being built as a subproject the tests won't be build, so I'm not sure we need an option here.  It's not like we're building a shared library or something; the *only* thing to build is the tests, so why would anyone want to "build" SIMDe without building the tests?

The only thing I can think of is that you could install the headers without building the tests, but that doesn't seem particularly useful…  Do you need this feature, or is it just the pattern you use for Meson projects?

Comment (nemequ) [reply meson.build:9]: Oh, nevermind, I just realized: Gentoo.  I guess you want to have a simde package which doesn't actually build anything and only installs to cut down on build times.

Comment (SoapGentoo) [reply meson.build:9]: Correct, we need to be able to disable tests. 95% of Gentoo users don't have tests enabled.

**Comment by nemequ** [comment]
Merged, thanks!

**Pull request closed without merging**